src/main/java/wolox/training/controllers/BookController.java

```diff
@@ -51,7 +51,8 @@ public Book findOne(@PathVariable Long id) {
 	@PostMapping
 	@ResponseStatus(HttpStatus.CREATED)
 	public Book create(@RequestBody Book book) {
-		bookRepository.findById(book.getId()).orElseThrow(() -> new BookNotFoundException("Book Not Found", new Exception()));
+		if(bookRepository.findById(book.getId()) != null)
+			throw new BookIdMismatchException("Book already created", new Exception());
```

Comment:
Maybe add a specific Exception for this case, BookAlreadyExistsException :) and the same goes for Users